app/src/sharedTest/java/org/oppia/app/profile/PinPasswordActivityTest.kt

```diff
@@ -77,6 +79,15 @@ class PinPasswordActivityTest {
       .inject(this)
   }
 
+  @Test
+  fun testPinPasswordActivityWithAdmin_activityVisible_checkKeyboardVisibilityByDefault() {
+    ActivityScenario.launch<PinPasswordActivity>(PinPasswordActivity.createPinPasswordActivityIntent(context, adminPin, userId)).use {
+      val inputMethodManager =
+        InstrumentationRegistry.getInstrumentation().targetContext.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+      assert(inputMethodManager.isAcceptingText)
```

Comment:
The code that you have written is correct. 
The issue is that the keyboard animation takes time because of which `assert` checks before that animation is finished as a result the test case fails. So, if you add a `SystemClock.sleep(500)` just before `assert`, you will notice that the test case pass, because by that time the keyboard animation has finished.

But actually, rather than checking if keyboard is visible or not, lets check if the `input_pin` has focus or not. That would solve the issue as well as make sure that the keyboard is connected to the view also.

`onView(withId(R.id.input_pin)).check(matches(hasFocus()))`

Reply:
Also, make sure that you remove unused imports when you remove the previous code.

Reply:
Thank you for suggesting, I have updated my test case code with the above code and also removed any unused imports.